### PR TITLE
Add Nix flake

### PR DESCRIPTION
**Pull request by siraben**
```ShellSession
$ NIXPKGS_ALLOW_UNFREE=1 nix build github:siraben/mcb/flake --impure
```

### CODE REVIEW

**Comment by siraben** [comment]
@gares ping

**Comment by gares** [comment]
Sorry I missed this PR.
I don't know what flake is, any pointer is welcome.
One thing that is a bit scary is the `--experimental-features` flag, if the code you wrote breaks because of this feature being experimental, I won't be able to fix it since I'm no nix expert.

**Comment by siraben** [comment]
Flakes are in stable Nix (just not currently enabled by default hence the flag). The main purpose it solves is making sure builds are hermetic and giving a standard interface to compose Nix projects. It's also backwards compatible with previous Nix versions (via https://github.com/edolstra/flake-compat/), so non-flakes users can continue using it fine. From a maintenance perspective the only command to run is `nix flake update` to keep the nixpkgs commit recent.

**Comment by siraben** [comment]
Note that the previous usage of Nix in this repository had an impure reference `with import <nixpkgs> {};` so this also improves reproducibility.

File: README.md [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
 [![build](https://github.com/math-comp/mcb/actions/workflows/build.yml/badge.svg)](https://github.com/math-comp/mcb/actions/workflows/build.yml)
 
-To build the book using Nix, run `nix-build`.  Alternatively you may
-fetch the latest artifact produced by the CI for the master branch
+## Building
+To build the book using Nix, run
+```ShellSession
+# without flakes, check out the repo first
+$ NIXPKGS_ALLOW_UNFREE=1 nix-build
+$ with flakes
```
Comment (gares) [inline README.md:9]: shouldn't this line start with #

Comment (gares) [inline README.md:9]:
```suggestion
# with flakes
```

**Comment by siraben** [comment]
Thanks!

**Pull request merged**